Elf/__init__.py

```diff
@@ -907,12 +900,44 @@ def getDynSymTabInfo(self):
         in self.dynsymtabct.  Perhaps this is horrible and should be stricken
         from the code.
         '''
-        #FIXME: make this use both SECTION and PT_DYNAMICS versions...
+        # if DT_SYMTAB doesn't exist, we don't have a symbol table defined by DYNAMICS
         symtabva = self.dyns.get(DT_SYMTAB)
         if symtabva is None:
             return None, None, None
 
         symsz = self.dyns.get(DT_SYMENT)
+
+        # calculate/reuse symbol count
+        if not self.dynsymtabct:    # check if we've already got this value
```

Comment:
It feels odd to check and set this multiple times and nest things so heavily. Do you think it'd be clearer to check for existance and bail early instead?

Reply:
not in Elf files.
there may or there may not.  and we may access things differently and we want to make sure this is accurate.